### ISSUE TITLE
Request for guidance

### ISSUE THREAD

**Issue by rkirandas** [hyperledger/aries-framework-dotnet#225]
Is there any walkthrough or guide to setup a cloud agent using Aries dotnet to create an issuer like Faber college / a verifier like acme and the Aries Max as Alice. And also the ledger network should be Sovrin Buildernet.

Any help would be appreciated

**Comment by x0axz** [comment]
[Yoroitchi](https://github.com/Yoroitchi) has created a fantastic Aries POV project called [AriesWebApp](https://github.com/Yoroitchi/AriesWebApp). It consists of three agents: the [Holder Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Holder5000), the [Issuer Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Issuer7000), and the [Verifier Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Verifier8000), all of which are developed in ASP.NET MVC and can easily be migrated to ASP.NET Web APIs.

**Comment by rkirandas** [comment]
@x0axz  Thank you . Do you know any goto person for provisioning based on sovrin net?

**Comment by x0axz** [comment]
@WadeBarnes @swcurran

**Comment by WadeBarnes** [comment]
@x0axz, @rkirandas, A simpler option for running your own local indy-node network instance is [von-network](https://github.com/bcgov/von-network).  It can be run on a local or remote machine for test purposes.  The Sovrin networks are great option when you get past the initial learning stage and you have something you want to expand; https://sovrin.org/overview/

**Comment by rkirandas** [comment]
> [Yoroitchi](https://github.com/Yoroitchi) has created a fantastic Aries POV project called [AriesWebApp](https://github.com/Yoroitchi/AriesWebApp). It consists of three agents: the [Holder Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Holder5000), the [Issuer Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Issuer7000), and the [Verifier Agent](https://github.com/Yoroitchi/AriesWebApp/tree/Verifier8000), all of which are developed in ASP.NET MVC and can easily be migrated to ASP.NET Web APIs.
> 
> In your Aries Max application, you might use the controller code from Holder Agent.
> 
> You don't need anything special for Mediator; simply use this [project](https://github.com/hyperledger/aries-mobile-agent-xamarin/tree/main/mediator).
> 
> I haven't tried the Sovrin Buildernet, so I can't comment on it. However, if you want to run your own network on a certain IP address, I recommend following this [instructions](https://github.com/hyperledger/indy-sdk#2-starting-the-test-pool-on-a-specific-ip-address). This [document](https://github.com/Yoroitchi/AriesWebApp/tree/Issuer7000#setup-a-demo) can also be used to deploy agents on servers.

@x0axz  

Do you knpwto configure the Web App to run on local ip instead of 192.168.56.101 explicit

**Comment by x0axz** [comment]
Here is the guide: https://github.com/hyperledger/indy-sdk#1-starting-the-test-pool-on-localhost

_To build a docker on localhost, use 127.0.0.1_
`$docker build --build-arg pool_ip=127.0.0.1 -f ci/indy-pool.dockerfile -t indy_pool .`

_To run a docker container on localhost, use 0.0.0.0. _
`$docker run -itd -p 0.0.0.0:9701-9708:9701-9708 indy_pool`

Note: Check the urls in [Startup.cs](https://github.com/Yoroitchi/AriesWebApp/blob/Issuer7000/AriesWebApp/Startup.cs#L41) and [Properties/launchSettings.json](https://github.com/Yoroitchi/AriesWebApp/blob/Issuer7000/AriesWebApp/Properties/launchSettings.json#L9) to make sure they're correct.

**Comment by rkirandas** [comment]
@x0axz  
should I change AriesTest.txn content to von-network and how about   

builder.RegisterAgent(option =>
                {
} section in Startup.cs


Have you done von-network version instead of indy

**Comment by rkirandas** [comment]
@x0axz 
One more thing  the  instructions of WebApp [https://github.com/Yoroitchi/AriesWebApp](url) is based on ubuntu  do you know how to setup it in WIndows?

**Comment by x0axz** [comment]
No, I haven't used Von Network, but yes, changes to the Genesis file are required (Aries.txn).

**Comment by rkirandas** [comment]
@x0axz  I ran the step 8 in https://github.com/Yoroitchi/AriesWebApp   Setup Demo but
docker run -it -p {port}:{port} --network=host {actor}         this command is not working in 

I tried to run an Issuer (7000) port    by 
 docker run -it -p {port}:7000 --network=host issuer123
showed error  docker: Invalid hostPort: {port}.

Also  tried docker run -it -p {port}=7000 --network=host issuer123 
showed error docker: No port specified: :/tcp<empty>.

I raised an issue in https://github.com/Yoroitchi/AriesWebApp Github but  no reply

**Comment by x0axz** [comment]
Replace port with `{port}` e.g. `docker run -it -p 7000=7000 --network=host issuer123`

**Comment by rkirandas** [comment]
@x0axz 
[https://github.com/Yoroitchi/AriesWebApp/issues/4](url)

**Comment by rkirandas** [comment]
@x0axz  Were you able to execute the issuer-verifier-holder demo  using  https://github.com/Yoroitchi/AriesWebApp

I got an error an Verifier Side mentioned https://github.com/hyperledger/aries-framework-dotnet/issues/4

**Comment by x0axz** [comment]
@https://github.com/Yoroitchi/AriesWebApp/issues/4#issuecomment-1149517976
